File: base/BinaryItem.php
<?php

namespace ffan\dop;

/**
 * Class ItemBinary
 * @package ffan\dop
 */
class BinaryItem extends Item
{
    /**
     * @var int 类型
     */
    protected $type = ItemType::BINARY;

    /**
     * 设置默认值
     * @param string $value
     * @return void
     * @throws DOPException
     */
    public function setDefault($value)
    {
        throw new DOPException('Binary 不支持二进制');
    }
}

File: test/java/DopTest/GsonTest/src/com/ffan/dop/BinaryItem.java
package com.ffan.dop;

import java.util.Base64;

class BinaryItem extends Item {
    /**
     * 值
     */
    private byte[] value;

    /**
     * 构造函数
     */
    BinaryItem(byte[] value) {
        this.value = value;
        this.type = ItemType.BINARY_TYPE;
    }

    /**
     * 获取值 byte
     */
    public byte[] getValueByte() {
        return  this.value;
    }

    /**
     * 获取值 string
     */
    public String getValueString() {
        return Base64.getEncoder().encodeToString(this.value);
    }
}